newIDE/app/src/MainFrame/Preferences/PreferencesDialog.js

```diff
@@ -63,6 +67,16 @@ const PreferencesDialog = ({ i18n, onClose }: Props) => {
     setShowEventBasedObjectsEditor,
     setNewProjectsDefaultFolder,
   } = React.useContext(PreferencesContext);
+  const [newProjectsFolder, setNewProjectsFolder] = React.useState(
+    values.newProjectsDefaultFolder
+  );
+
+  const onRequestClose = () => {
+    setNewProjectsDefaultFolder(
```

Comment:
Is this:
- the responsibility of the poor PreferencesDialog to deal with this? :)
- even necessary at all? ;)

Reply:
Well, I think that it's necessary.

```typescript
  const onRequestClose = () => {
    setNewProjectsDefaultFolder(
      newProjectsFolder === '' ? findDefaultFolder(app) : newProjectsFolder
    );
    onClose(languageDidChange);
  };
```

On closing, we check the input's value, and if empty, we set it with the default folder. I don't see the problem.
Am I missing something?